File: Ej2extra.java
package ej2extra;

import javax.swing.JOptionPane;

class Ej2extra {

	public static void main(String[] args) {
		String nom;
		int anac, edad, edf, aa;
		
		nom=JOptionPane.showInputDialog("ingrese nombre");
		anac=Integer.parseInt(JOptionPane.showInputDialog("ingrese ano nacimiento"));
		aa=Integer.parseInt(JOptionPane.showInputDialog("ingrese ano actual"));	
		edad=aa-anac;
		edf=2050-anac;
		
		JOptionPane.showMessageDialog(null, nom + " su edad es: " + edad + "\nen el 2050: " + edf);
	}
}

File: Ej2extra.js
let nom = prompt("Ingrese nombre");
let anac = parseInt(prompt("Ingrese año nacimiento"));
let aa = parseInt(prompt("Ingrese año actual"));

let edad = aa - anac;
let edf = 2050 - anac;

alert(nom + " su edad es: " + edad + "\nEn el 2050: " + edf);
